File: 0958-check-completeness-of-a-binary-tree/0958-check-completeness-of-a-binary-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left),
 * right(right) {}
 * };
 */
class Solution {
public:
    bool isCompleteTree(TreeNode* root) {
        queue<TreeNode*> q;
        q.push(root);
        // apply BFS
        bool flag = false;
        while (q.size() > 0) {
            int length = q.size();
            for (int i = 0; i < length; i++) {
                cout<<flag;
                TreeNode* temp = q.front();
                q.pop();
                if(temp==NULL){
                    flag = true;
                    continue;
                }
                if(flag==true){
                    return false;
                }
                
                TreeNode* left = temp->left;
                TreeNode* right = temp->right;
                q.push(left);
                q.push(right);
               
                
            }
            
        }
        return true;
    }
};

File: 0958-check-completeness-of-a-binary-tree/0958-check-completeness-of-a-binary-tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {boolean}
 */
var isCompleteTree = function (root) {
    const queue = []
    queue.push(root)
    var isNullpresent = false;
    while (queue.length > 0) {
        let length = queue.length;
        console.log({ queue })
        for (let i = 0; i < length; i++) {
            let temp = queue[0];
            // console.log(temp.val)
            queue.shift();
            if (temp != null) {
                if (isNullpresent){
                    return  false
                }
                queue.push(temp.left)
                queue.push(temp.right)
            }
            else {
                isNullpresent = true;
            }



        }
    }
    return true

};